src/Workspaces/Remote/ServiceHub/Shared/ServiceHubServiceBase.cs

```diff
@@ -29,6 +29,7 @@ internal abstract class ServiceHubServiceBase : IDisposable
         private Checksum _solutionChecksumOpt;
         private RoslynServices _lazyRoslynServices;
 
+        [Obsolete("For backward compatibility. this will be removed once all callers moved to new ctor")]
```

Comment:
What existing callers are using this?  Why not move them to the new constructor as well?